client/my-sites/stats/controller.jsx

```diff
@@ -115,24 +112,16 @@ export default {
 		context.store.dispatch( savePreference( 'firstViewHistory', [] ) );
 	},
 
-	redirectToDefaultSitePage: function( context, next ) {
+	redirectToDefaultSitePage: function( context ) {
 		const siteFragment = getSiteFragment( context.path );
 
-		if ( siteFragment ) {
```

Comment:
Do we not need to check for this anymore?

Reply:
I don't think so, because I want to always redirect when that controller is called - previously it was more of a _maybe_ redirect, which also caused the issue by calling `next()` when `/stats/(.*)` was matched.